pkg/generate/app/cmd/newapp.go

```diff
@@ -325,10 +337,17 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 				if err != nil {
 					return nil, fmt.Errorf("can't build %q: %v", ref.Input(), err)
 				}
-				strategy, source, err := app.StrategyAndSourceForRepository(ref.Input().Uses)
+				strategy, source, err := app.StrategyAndSourceForRepository(ref.Input().Uses, app.PerfectMatchWeightedResolver{
```

Comment:
Oh yea, I have changed the weight in the docker resolver. Well, I am not sure about it to be honest. @csrwng is doing the same in [generate](https://github.com/openshift/origin/blob/09f87e4d4e975d96948900ed6a8edee4365bc380/pkg/cmd/experimental/generate/generate.go#L218)